pymeasure/instruments/hp/hp6641a.py

```diff
@@ -0,0 +1,199 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import enum
+from pymeasure.instruments import Instrument, SCPIUnknownMixin
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+class OperationStatus(enum.Flag):
+    """Operation Status."""
+    CAL = 1 << 0
+    WTG = 1 << 5
+    CV = 1 << 8
+    CC = 1 << 10
+
+
+class QuestionableStatus(enum.Flag):
+    """Questionable Status."""
+    OV = 1 << 0
+    OC = 1 << 1
+    OT = 1 << 4
+    RI = 1 << 9
+    UNR = 1 << 10
+
+
+class StandardEventStatus(enum.Flag):
+    """Standard Event Status."""
+    OPC = 1 << 0
+    QYE = 1 << 2
+    DDE = 1 << 3
+    EXE = 1 << 4
+    CME = 1 << 5
+    PON = 1 << 7
+
+
+limits = {
+    "HP6641A": {"Volt_lim": 8.190, "OVP_lim": 8.8, "Cur_lim": 20.475},
+    "HP6673A": {"Volt_lim": 35.831, "OVP_lim": 42.0, "Cur_lim": 61.43},
+    "HP6674A": {"Volt_lim": 61.425, "OVP_lim": 72.0, "Cur_lim": 35.83}}
+
+
+class HP6641A(SCPIUnknownMixin, Instrument):
+    """ Represents the HP / Agilent 6641A
+    provides a high-level interface for interacting with the instrument.
+
+    .. code-block:: python
+
+        instr =  vxi11.Instrument("192.168.88.116", "gpib0,6")
+        psu = hp6641a.HP6641A(instr);
+        psu.clear()
+        psu.remote_control_enabled = True
+        psu.set_voltage = 3.3
+        psu.set_current = 1.5
+        psu.set_ovp = 8.0
+        psu.output = 1
+        print(f'read errors: {psu.pop_err()}')
+        while True:
+            print(f'output voltage: {psu.meas_voltage}')
+            print(f'output current: {psu.meas_current}')
+            time.sleep(1)
+    """
+
+    BOOL_MAPPINGS = {True: 1, False: 0}
+    DISP_MAPPINGS = {True: "TEXT", False: "NORM"}
+
+    def __init__(self, adapter, name="HP6641A", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    set_voltage = Instrument.control(
+        "VOLT?", "VOLT %g",
+        "Set and Get the voltage programmed level",
+        dynamic=True,
+        validator=strict_range,
+        values=[0, limits["HP6641A"]["Volt_lim"]]
+    )
+
+    set_ovp = Instrument.control(
+        "VOLT:PROT?", "VOLT:PROT %g",
+        "Set and Get overvoltage protection and programmed level",
+        dynamic=True,
+        validator=strict_range,
+        values=[0, limits["HP6641A"]["OVP_lim"]]
+    )
+
+    set_current = Instrument.control(
+        "CURR?", "CURR %g",
+        "Set and Get the current programmed level",
+        dynamic=True,
+        validator=strict_range,
+        values=[0, limits["HP6641A"]["Cur_lim"]]
+    )
+
+    meas_voltage = Instrument.measurement(
+        "MEAS:VOLT?",
+        "Measure the voltage at the power supply's sense terminals"
+    )
+
+    meas_current = Instrument.measurement(
+        "MEAS:CURR?",
+        "Measure the current at the power supply's sense terminals"
+    )
+
+    output = Instrument.control(
```

Comment:
For a boolean, it is good to name it `output_enabled` to make it clear what `output_enabled = True` means.